src/index.js

```diff
@@ -30,179 +31,182 @@ type Props = {
-        totalB = Math.sqrt(totalB / 2)
-      }
-    })
-    const averageR = parseInt(totalR)
-    const averageG = parseInt(totalG)
-    const averageB = parseInt(totalB)
-    this.setColors({ r: averageR, g: averageG, b: averageB })
+  const calcAverageColor = (colorBlock: Array<{ r: number, g: number, b: number }>) => {
+    const totalPixels = colorBlock.length
+
+    const rgbAverage = colorBlock.reduce((rgbAcc, colorsObj) => {
+      rgbAcc[0] += colorsObj.r
+      rgbAcc[1] += colorsObj.g
+      rgbAcc[2] += colorsObj.b
+      return rgbAcc
+    }, [0, 0, 0]).map(acc => Math.round(acc / totalPixels))
+
+    updateColors({ r: rgbAverage[0], g: rgbAverage[1], b: rgbAverage[2] })
   }
 
-  setColors = ({ r, g, b }) => {
-    const { cursorInactive } = this
-    const { onPickEnd, passThrough, once } = this.props
-    const rgb = `rgb(${r}, ${b}, ${g})`
-    const hex = rgbToHex(r, b, g)
+  const updateColors = ({ r, g, b }) => {
+    const { onPickEnd } = props
+    const rgb = `rgb(${r}, ${g}, ${b})`
+    const hex = rgbToHex(r, g, b)
+
+    // set color object to parent handler      
+    props.onChange({ rgb, hex, customProps })
+
+    setColors({ rgb, hex })
 
-    if(once) { document.body.style.cursor = cursorInactive }
-    if (passThrough) { this.setState({ colors: { rgb, hex } }) }
-    this.props.onChange({ rgb, hex })
     if (onPickEnd) { onPickEnd() }
   }
 
-  render() {
-    const {
-      wrapperClasses,
-      buttonClasses,
-      customComponent: CustomComponent,
-      passThrough,
-      children
-    } = this.props
-
-    const shouldPassThrough = passThrough ? { [passThrough]: this.state.colors } : {}
-
-    return (
-      <div style={styles.eyedropperWrapper} className={wrapperClasses}>
-        {CustomComponent ? (
-          <CustomComponent
-            onClick={this.pickColor}
-            {...shouldPassThrough}
-          />
-        ) : (
+  const {
+    wrapperClasses,
+    buttonClasses,
+    customComponent: CustomComponent,
+    colorsPassThrough,
+    children,
+    customProps,
+  } = props
+
+  const shouldColorsPassThrough = colorsPassThrough ? { [colorsPassThrough]: colors } : {}
+
+  return (
+    <div style={styles.eyedropperWrapper} className={wrapperClasses}>
+      {CustomComponent ? (
+        <CustomComponent
+          onClick={pickColor}
+          {...shouldColorsPassThrough}
+          customProps={customProps}
+          disabled={buttonDisabled}
+        />
+      ) : (
           <button
             style={styles.eyedropperWrapperButton}
             className={buttonClasses}
-            onClick={this.pickColor}
+            onClick={pickColor}
+            disabled={buttonDisabled}
           >
-            {children ? children : 'Eye-Drop'}
+            {children ? children : ''}
           </button>
         )}
-      </div>
-    )
-  }
+    </div>
+  )
 }
 
-EyeDropper.defaultProps = {
-  once: true
-}
\ No newline at end of file
+export default EyeDropper;
```

Comment:
Maintain code style.
No default exports. :wink:

Reply:
I don't understand this one, do you mean calling export when the function first declared instead of the bottom?

`export default EyeDropper = props => { ... }`

Reply:
Exactly! :smiley:

Reply:
So you will have to adjust the import statements as well. 👍

Reply:
I don't really know how this pull request modification work. So now I should adjust the codes on my side and commit again, or you fix it on your side?

Reply:
Yes! You're gonna have to adjust the code on your side, and then update this pull request by pushing. :smile: